### DISCUSSION TITLE
Is there a way to click a message on chat?

### DISCUSSION THREAD

**Discussion by v4ndn** [PrismarineJS/mineflayer#2679] [Q&A] [answered]
As i know, hypixel have random command on skyblock when you trying to coopadd a player, i need to click on chat then click on accept button

How it looks:
First i click on message
![изображение](https://user-images.githubusercontent.com/59515058/176226791-2aef9b1d-336e-4ac3-9524-9cd03919486e.png)

Then click accept (Green clay)
![изображение](https://user-images.githubusercontent.com/59515058/176226876-ecb8698b-8fc2-41f3-a9e8-4c49d31e6515.png)

Is there a way to do this?

**Comment by amoraschi** [comment, marked as answer]
I think you can get the JSON message from the "message" event, and it comes with a string that is basically the command that gets executed when you click the message.

As for the window clicking thing:
```
All of the following assumes that you have already opened the desired window.
To click on an item in a window use: bot.clickWindow(<slot>, <mouse_button>, <mode>)

This will return a Promise of type void, which means it can be awaited.
<slot> is the item.slot of the item you want to click on in the window.
<mouse_button> is 0 for left click, and 1 for right click.
<mode> is the mode of the click, mineflayer only supports 0 for this.
To get more info:
```
https://github.com/PrismarineJS/mineflayer/blob/master/docs/api.md#botclickwindowslot-mousebutton-mode

**Reply by v4ndn** [reply]
ok, how i can get the opened window? i tried just use bot.currentWindow, but it returns my inventory
or i need just use click?

![изображение](https://user-images.githubusercontent.com/59515058/176279329-b1b8d1b3-166f-4171-afc4-817e23ab353b.png)
![изображение](https://user-images.githubusercontent.com/59515058/176279370-8db01064-2048-49ff-bb7a-ee996792d83d.png)

**Reply by amoraschi** [reply]
You can wait for a window to open with:
https://github.com/PrismarineJS/mineflayer/blob/master/docs/api.md#windowopen-window
